lib/api/CPersistenceManager.cc

```diff
@@ -172,13 +178,24 @@ bool CBackgroundPersister::startBackgroundPersistIfAppropriate() {
         return false;
     }
 
-    return this->startBackgroundPersist(now);
+    return this->startPersist(now);
 }
 
-bool CBackgroundPersister::startBackgroundPersist(core_t::TTime timeOfPersistence) {
-    bool backgroundPersistSetupOk = m_FirstProcessorPeriodicPersistFunc(*this);
-    if (!backgroundPersistSetupOk) {
-        LOG_ERROR(<< "Failed to create background persistence functions");
+bool CPersistenceManager::startPersist(core_t::TTime timeOfPersistence) {
+    if (this->isBusy()) {
+        LOG_WARN(<< "Cannot start persist as a previous "
+                    "background persist is still in progress");
+        return false;
+    }
+
+    auto firstProcessorPeriodicPersistFunc = [&]() {
+        return m_PersistInForeground ? m_FirstProcessorForegroundPeriodicPersistFunc
+                                     : m_FirstProcessorBackgroundPeriodicPersistFunc;
+    };
```

Comment:
nit: why wrap in a lambda? Surely this could just be
```
auto firstProcessorPeriodicPersistFunc = m_PersistInForeground
                                     ? m_FirstProcessorForegroundPeriodicPersistFunc
                                     : m_FirstProcessorBackgroundPeriodicPersistFunc
```